src/linux/osd.c

```diff
@@ -64,3 +70,100 @@ ssize_t ofi_get_hugepage_size(void)
 
 	return val * 1024;
 }
+
+#ifdef HAVE_ETHTOOL
+static inline size_t ofi_ethtool_get_speed(struct ethtool_cmd *ecmd)
+{
+	uint32_t speed_mbps;
+
+#ifdef HAVE_ETHTOOL_CMD_SPEED
+	speed_mbps = ethtool_cmd_speed(ecmd);
+#else /* HAVE_ETHTOOL_CMD_SPEED */
+	speed_mbps = ecmd->speed;
+#endif /* HAVE_ETHTOOL_CMD_SPEED */
+
+	/* If the value of the speed is unknonw, set it to 100 Mb/s */
+#ifdef HAVE_ETHTOOL_SPEED_UNKNOWN
+	speed_mbps = (speed_mbps != SPEED_UNKNOWN) ? speed_mbps : 100;
+#else /* HAVE_ETHTOOL_SPEED_UNKNOWN */
+	speed_mbps = ((speed_mbps != 0) &&
+		       ((uint16_t)speed_mbps != (uint16_t)-1)) ?
+		     speed_mbps : 100;
+#endif /* HAVE_ETHTOOL_SPEED_UNKNOWN */
+
+	return speed_mbps;
+}
+#endif /* HAVE_ETHTOOL */
```

Comment:
Please avoid ifdefs in the middle of functions.  Create separate functions instead.  Applies here and below.

Reply:
Done